tests/servo_test.py

```diff
@@ -701,11 +706,11 @@ async def test_aliased_connectors_produce_schema(
                     "env_names": [
                         "SERVO_OPTIMIZER",
                     ],
-                    "allOf": [
-                        {
-                            "$ref": "#/definitions/Optimizer",
-                        },
+                    "anyOf": [
```

Comment:
I looked into it and the TLDR seems to be "No"

1. The main difference between anyOf and oneOf is that the former is OR and the latter is XOR. https://json-schema.org/understanding-json-schema/reference/combining.html#oneof
    1. The models being Unioned in the config are already mutually exclusive by differences in their fields as there needs to be some difference between them in order for the config parsing to work as expected. We don't really expect to have cases where the fields of the config types completely overlap (if in the future we do have such cases then we would need to apply a discriminator to their model  https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions )
    2. Pydantic itself does not enforce the XOR constraint on Unions and simply returns an instance of the first object to be successfully parsed. Therefore the authors feel it would be a misnomer to use oneOf in the schematization. https://github.com/pydantic/pydantic/issues/656